src/programs/azimuthal_average/azimuthal_average.cpp

```diff
@@ -0,0 +1,1534 @@
+		{
+
+			input_text->ReadLine(temp_float);
+
+			if (input_text->records_per_line == 3)
+			{
+				ctf_parameters[image_counter][0] = acceleration_voltage;
+				ctf_parameters[image_counter][1] = spherical_aberration;
+				ctf_parameters[image_counter][2] = amplitude_contrast;
+				ctf_parameters[image_counter][3] = temp_float[0];
+				ctf_parameters[image_counter][4] = temp_float[1];
+				ctf_parameters[image_counter][5] = temp_float[2];
+				ctf_parameters[image_counter][6] = 0.0;
+				ctf_parameters[image_counter][7] = 0.5;
+				ctf_parameters[image_counter][8] = 0.0;
+				ctf_parameters[image_counter][9] = pixel_size;
+				ctf_parameters[image_counter][10] = 0.0;
+			}
+			else
+			{
+				ctf_parameters[image_counter][0] = acceleration_voltage;
+				ctf_parameters[image_counter][1] = spherical_aberration;
+				ctf_parameters[image_counter][2] = amplitude_contrast;
+				ctf_parameters[image_counter][3] = temp_float[0];
+				ctf_parameters[image_counter][4] = temp_float[1];
+				ctf_parameters[image_counter][5] = temp_float[2];
+				ctf_parameters[image_counter][6] = 0.0;
+				ctf_parameters[image_counter][7] = 0.5;
+				ctf_parameters[image_counter][8] = 0.0;
+				ctf_parameters[image_counter][9] = pixel_size;
+				ctf_parameters[image_counter][10] = temp_float[3];
+			}
+		}
+	}
+
+	// read in image stack and FFT
+
+	read_frames_start = wxDateTime::Now();
+	// input stack times ctf
+	Image *image_stack_times_ctf = new Image[number_of_input_images];
+	// allocate arrays for the ctf sum of squares
+	float *ctf_sum_of_squares;
+
+	// read image stack
+
+	for (image_counter = 0; image_counter < number_of_input_images; image_counter++)
+	{
+		// Read from disk
+		image_stack[image_counter].ReadSlice(&my_input_file, image_counter + 1);
+		image_stack_times_ctf[image_counter].ReadSlice(&my_input_file, image_counter + 1);
+
+		// Normalize images
+		normalize_image(&image_stack[image_counter], pixel_size, 10.0);
+		normalize_image(&image_stack_times_ctf[image_counter], pixel_size, 10.0);
+
+		// read CTF from file
+		if (input_ctf_values_from_text_file == true)
+		{
+			current_ctf.Init(ctf_parameters[image_counter][0], ctf_parameters[image_counter][1], ctf_parameters[image_counter][2], ctf_parameters[image_counter][3], ctf_parameters[image_counter][4], ctf_parameters[image_counter][5], ctf_parameters[image_counter][6], ctf_parameters[image_counter][7], ctf_parameters[image_counter][8], ctf_parameters[image_counter][9], ctf_parameters[image_counter][10]);
+		}
+
+		// allocate arrays for the ctf sum of squares
+		if (image_counter == 0)
+		{
+			ctf_sum_of_squares = new float[image_stack[0].real_memory_allocated / 2];
+			ZeroFloatArray(ctf_sum_of_squares, image_stack[0].real_memory_allocated / 2);
+		}
+
+		// apply CTF
+		image_stack_times_ctf[image_counter].ForwardFFT();
+		image_stack_times_ctf[image_counter].ZeroCentralPixel();
+		apply_ctf(&image_stack_times_ctf[image_counter], current_ctf, ctf_sum_of_squares, phase_flip_only);
```

Comment:
I haven't checked this method yet, but I assume this adds some functionality not in the existing Image class method?

Reply:
The method ApplyCTF in image.cpp does not store the ctf_sum_of_squares to correct for the CTF. The purpose of creating my own function was to add this functionality--the way the ctf is applied to each image is exactly the same in both methods--and make the azimuthal average method self-contained as much as possible.

Reply:
So you could do this in a couple of lines:

void apply_ctf {
Image buffer;
buffer.CalculateCTFImage( args)
image_stack_times_ctf.MultiplyPixelWise(buffer)
buffer.MultiplyPixelWise(buffer)
ctf_sum_of_squares.AddImage(buffer)
}